### DISCUSSION TITLE
day-8, level 3 exercises q2

### DISCUSSION THREAD

**Discussion by KhuramShahzad-5335** [Asabeneh/30-Days-Of-JavaScript#666] [Q&A] [answered]
i need help with the sign in and signup function for day8, level3,exercise q2,

**Comment by anchalpaswan** [comment, marked as answer]
function signUp(username, email, password) {
   
    //Check if the user already exists
    const existingUser = users.find(user => user.email === email);
    if (existingUser) {
      console.log('User with this email already exists!');
      return;
    }
    // Create a new user object with a unique ID and current time as createdAt
    const newUser = {
      _id: Math.random().toString(36).substring(2),
      username,
      email,
      password,
      createdAt: new Date().toLocaleString(),
      isLoggedIn: false
    };
    // Add the new user to the array
    users.push(newUser);
    console.log('User successfully signed up:', newUser);
  }
  
  function signIn(email, password) {
    // Find the user with the given email and password
    const user = users.find(user => user.email === email && user.password === password);
    if (user) {
      // Update the user's isLoggedIn flag to true and print a success message
      user.isLoggedIn = true;
      console.log('User successfully signed in:', user);
    } else {
      console.log('Invalid email or password!');
    }
  }

**Reply by KhuramShahzad-5335** [reply]
Thanks😃
